From ca036970b0e920a1d80140195558487f1c5346a0 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@st.com>
Date: Fri, 16 Apr 2021 16:31:08 +0200
Subject: [PATCH 39/43] ARM: dts: stm32: Add support to retrieve Ethernet @MAC
 with nvmem on stm32mp13

MAC1 and MAC2 are fused on OTP (57/58), two mechanisms to retrieve them:
- use "aliases": bootloader retrieve it and set "local-mac-address",
which is used during probe to set MAC address
- if "aliases" not set for ethernet, nvmem read directly OTP fused at
probe to set MAC address

Upstream-Status: Inappropriate [ST Downstream https://github.com/STMicroelectronics/linux.git 26f146964c13d8043ef027bf059befacdd1cd148]
Signed-off-by: Christophe Roullier <christophe.roullier@st.com>
Change-Id: Iad564edd07682d7ad6e7ef0e529987180d22c14e
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/229830
Reviewed-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-by: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Tested-by: Eric FOURMONT <eric.fourmont-ext@st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/284374
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
Domain-Review: Alexandre TORGUE <alexandre.torgue@foss.st.com>
---
 arch/arm/boot/dts/st/stm32mp131.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/boot/dts/st/stm32mp131.dtsi b/arch/arm/boot/dts/st/stm32mp131.dtsi
index 27842045f0e86..179aa3a17c381 100644
--- a/arch/arm/boot/dts/st/stm32mp131.dtsi
+++ b/arch/arm/boot/dts/st/stm32mp131.dtsi
@@ -1403,6 +1403,12 @@ ts_cal1: calib@5c {
 			ts_cal2: calib@5e {
 				reg = <0x5e 0x2>;
 			};
+			ethernet_mac1_address: mac1@e4 {
+				reg = <0xe4 0x6>;
+			};
+			ethernet_mac2_address: mac2@ea {
+				reg = <0xea 0x6>;
+			};
 		};
 
 		/*
-- 
2.43.0

